File: Rearrange_array_alternately/code.cpp
class Solution{
    public:
    // This function wants you to modify the given input
    // array and no need to return anything
    // arr: input array
    // n: size of array
    //Function to rearrange  the array elements alternately.
    void rearrange(long long *arr, int n) 
    { 
    	
    	vector<long long> res;
    	for(int i=0; i<n/2; i++){
    	    res.push_back(arr[n-i-1]);
    	    res.push_back(arr[i]);
    	}
    	if(n%2!=0){
    	    res.push_back(arr[n/2]);
    	}
    	for(int i=0; i<n; i++){
    	    arr[i] = res[i];
    	}
    	 
    }
};

File: Rearrange_array_alternately/code.java


class Solution{
    
    // temp: input array
    // n: size of array
    //Function to rearrange  the array elements alternately.
    public static void rearrange(long arr[], int n){
        
        // Your code here
        ArrayList<Long> list=new ArrayList<Long>();
        int i=0;
        int j=n-1;
        while(i<j)
        {
            list.add(arr[j]);
            list.add(arr[i]);
            i++;
            j--;
        }
        if(i==j)
        {
            list.add(arr[i]);
        }
        
        for(int k=0;k<n;k++)
        {
            arr[k]=list.get(k);
        }


        
    }
    
}
